src/trading/NewOrders.sol

```diff
@@ -0,0 +1,338 @@
+        return orders[_orderId].fxpMoneyEscrowed;
+    }
+
+    function getVolume(IMarket _market) public constant returns (uint256) { 
+        return marketOrderData[_market].volume;
+    }
+
+    function getLastOutcomePrice(IMarket _market, uint8 _outcome) public constant returns (uint256) { 
+        return marketOrderData[_market].prices[_outcome];
+    }
+
+    function getBetterOrderId(bytes20 _orderId, Trading.TradeTypes _type, IMarket _market, uint8 _outcome) public constant returns (bytes20) { 
+        return orders[_orderId].betterOrderId;
+    }
+
+    function getWorseOrderId(bytes20 _orderId, Trading.TradeTypes _type, IMarket _market, uint8 _outcome) public constant returns (bytes20) {
+        return orders[_orderId].worseOrderId;
+    }
+
+    function getGasPrice(bytes20 _orderId, Trading.TradeTypes _type, IMarket _market, uint8 _outcome) public constant returns (uint256) { 
+        return orders[_orderId].gasPrice;
+    }
+
+    function getBestOrderId(Trading.TradeTypes _type, IMarket _market, uint8 _outcome) public constant returns (bytes20) { 
+        return bestOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)];
+    }
+
+    function getWorstOrderId(Trading.TradeTypes _type, IMarket _market, uint8 _outcome) public constant returns (bytes20) { 
+        return worstOrder[getBestOrderWorstOrderHash(_market, _outcome, _type)];
+    }
+
+    // FIXME: Currently, getOrderIdHash(), getBestOrderWorstOrderHash(), and makeOrder.makeOrder() are creating ripemd160 hash values of type bytes20.  This should probably be changed to create sha3/sha256 hash values of type bytes32.
+    function getOrderIdHash(Trading.TradeTypes _type, IMarket _market, uint256 _fxpAmount, uint256 _fxpPrice, address _sender, uint256 _blockNumber, uint256 _outcome, uint256 _fxpMoneyEscrowed, uint256 _fxpSharesEscrowed) public constant returns (bytes20) {
+        return ripemd160(_type, _market, _fxpAmount, _fxpPrice, _sender, _blockNumber, _outcome, _fxpMoneyEscrowed, _fxpSharesEscrowed);
+    }
+
+    function getBlockNumber() public constant returns (uint256) {
+        return block.number;
+    }
+
+    function isBetterPrice(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _orderId) public constant returns (bool) { 
+        if (_type == Trading.TradeTypes.Bid) {
+            return (_fxpPrice > orders[_orderId].fxpPrice);
+        } else {
+            return (_fxpPrice < orders[_orderId].fxpPrice);
+        }
+    }
+
+    function isWorsePrice(Trading.TradeTypes _type, IMarket _market, uint8 _outcome, uint256 _fxpPrice, bytes20 _orderId) public constant returns (bool) { 
+        if (_type == Trading.TradeTypes.Bid) { 
+            return (_fxpPrice < orders[_orderId].fxpPrice);
+        } else {
+            return (_fxpPrice > orders[_orderId].fxpPrice);
+        }
```

Comment:
Style: I personally prefer switch statements for things like this as it ensures that all of the enum values are handled.  For a 2-item switch it doesn't really make a huge difference, but I think it adds some clarity.  Right now this code makes it look like the second codepath should be followed if _type is anything other than Bid, which isn't correct.

If you prefer to keep it as an if statement, this function should `require(_type == Trading.TradeTypes.Bid || _type == Trading.TradeTypes.Ask)` or it should be if, else if, else throw.  We want to protect ourselves from ourselves and someone passing something other than one of those two values into this function.

Reply:
I don't think Solidity actually supports switch statements (http://solidity.readthedocs.io/en/develop/control-structures.html#control-structures), but I'll change the `else` to an `else if` and add the `require()` line you mentioned above.

Reply:
Odd.  It appears that Solidity supports switch statements in inline assembly, but not in normal Solidity.  Very bizarre.

Reply:
Hmm, that is really weird.  Well, hopefully Solidity will add switch statements at some point in the future.